source/common/router/rds_subscription.cc

```diff
@@ -59,6 +59,7 @@ void RdsSubscription::parseResponse(const Http::Message& response) {
 void RdsSubscription::onFetchComplete() {}
 
 void RdsSubscription::onFetchFailure(const EnvoyException* e) {
+  RELEASE_ASSERT(callbacks_ != nullptr);
```

Comment:
Generally we don't do this as it will crash in an obvious way on the next line. Why is this required?

Reply:
I like to put in asserts like this to force the crash to happen as far up the call graph as possible.

Crashes from null dereference don't always localize well -- the final SIGSEGV can be triggered somewhere deep down in the stack, if no instruction needed to resolve `*this` immediately.

Reply:
In general this has not been my experience, and we don't currently do this anywhere else (see error handling / style guide). For consistency I would prefer to not do this. If we need it for coverity can add relevant comments?

Reply:
OK, removed these asserts of non-null callbacks, and added a comment to the dynamic_cast assertion.